src/paradigma/heart_rate/heart_rate_estimation.py

```diff
@@ -0,0 +1,185 @@
+import numpy as np
+from typing import Tuple
+
+from paradigma.config import HeartRateExtractionConfig
+from paradigma.heart_rate.tfd import nonsep_gdtfd
+
+def assign_sqa_label(ppg_prob, config: HeartRateExtractionConfig, acc_label=None) -> np.ndarray:
+    """
+    Assigns a signal quality label to every individual data point.
+
+    Parameters:
+    - ppg_prob: numpy array of probabilities for PPG
+    - acc_label: numpy array of labels for accelerometer (optional)
+    - fs: Sampling frequency
+
+    Returns:
+    - sqa_label: numpy array of signal quality assessment labels
+    """
+    # Default _label to ones if not provided
+    if acc_label is None:
+        acc_label = np.ones(len(ppg_prob))
+
+
+    # Number of samples in an epoch
+    fs = config.sampling_frequency
+    samples_per_epoch = config.window_length_s * fs
+
+    # Calculate number of samples to shift for each epoch
+    samples_shift = config.window_step_size_s * fs
+    n_samples = int((len(ppg_prob) + config.window_overlap_s) * fs)
+    data_prob = np.zeros(n_samples)
+    data_label_imu = np.zeros(n_samples, dtype=np.int8)
+
+    for i in range(n_samples):
+        # Start and end indices for current epoch
+        start_idx = int(np.floor((i - (samples_per_epoch - samples_shift)) / fs))
+        end_idx = int(np.floor(i / fs))
+
+        # Correct for first and last epochs
+        if start_idx < 0:
+            start_idx = 0
+        if end_idx > len(ppg_prob):
+            end_idx = len(ppg_prob)
+
+        # Extract probabilities and labels for the current epoch
+        prob = ppg_prob[start_idx:end_idx+1]
+        label_imu = acc_label[start_idx:end_idx+1]
+
+        # Calculate mean probability and majority voting for labels
+        data_prob[i] = np.mean(prob)
+        data_label_imu[i] = int(np.mean(label_imu) >= 0.5)
```

Comment:
Use round here as well if appropriate.

Reply:
In this case not, since it is a true false statement which I want to convert to 0 or 1, therefore this doesn't have to be rounded

Reply:
Haha you're right, not sure where this suggestion came from. You might want to store it as `boolean` if TSDF allows (don't think it does yet though ...)